Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,59 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
+        public static int FindDigit(string equation)
+        {         
             // Add your code here.
-            throw new NotImplementedException();
+            //Splitting The equation into three variables eg. 42*47=1?74 to A=42 B=47 and C=1?74
+            var A = equation.Split('*')[0];
+            var B = equation.Split('*')[1].Split('=')[0];
+            var C = equation.Split('=')[1];
+            var result = -1;
+            var ans = 0.0;
+            String originalValue = String.Empty, evaluatedValue;
+
+            //Checking which string variable contain "?" mark and storing the evaluated result in ans
+            if(A.Contains("?"))
+            {
+                ans = double.Parse(C)/double.Parse(B);
+                originalValue = A;
+            }
+            else if(B.Contains("?"))
+            {
+                ans = double.Parse(C)/double.Parse(A);
+                originalValue = B;
+            }
+            else if(C.Contains("?"))
+            {
+                ans = double.Parse(A) * double.Parse(B);
+                originalValue = C;
+            }
+            evaluatedValue = ans.ToString();
+
+            //Comparing the original and evaluated variables and finding the missing value
+            if(originalValue.Length == evaluatedValue.Length)
+            {
+                for(int i =0; i < evaluatedValue.Length; i++)

```

Comment:
It is comparing character by character. Isn't there an approach which is optimal?

Reply:
if(equation.Length == evaluatedEquation.Length)
            {
                int indexOfQuestionMark = equation.IndexOf('?');
                tempChar = evaluatedEquation[ indexOfQuestioNMark ];
                result = tempChar  - '0';
                return result;
            }
for this i cam up with this 
where  evaluatedEquation is a string which is same as the equation given
evaluatedEquation = A + "*" + B + "=" + C ;
where A,B,C are the evaluated value